src/models/navigator.ts

```diff
@@ -49,6 +49,7 @@ export interface ItemCustomization {
 
 export type SectionCustomComponentProps = {
   sectionInstance: SectionInstance;
+  sectionBlueprint?: SectionBlueprint<any>;
```

Comment:
Let's remove this from here because each `sectionInstance` has the same `id` as it's `sectionBlueprint`.
Thus, in your custom component you can import the `sectionBlueprintMap` and do the following to get the blueprint:
```const sectionBlueprint = sectionBlueprintMap.getById(sectionInstance.id);```

Reply:
All parent components have the blue print injected so it's a bit odd to rely on a global variable in the leaf component. My feeling is that injecting dependencies is easier to reason about than using globals. I'm curious to your reasoning behind using the map? (Do note that I do not mind changing it, trying to understand for the future)

Reply:
The blueprint object is not serializable and it probably won't work well when React checks if the blueprint sent as prop as changed. It could lead to unnecessary re-renders.

I don't see any problem in relying on global objects when it's something used in many places of the app. This is similar to the `store` object from Redux as which you could import in any file even if it's not a React component and you can still read the state or dispatch actions to it.

Reply:
And yes, it should be exceptional when those components need the blueprint.

We have `getMeta` on items if we need to create `any` object we want on the `instance.meta`

Reply:
That makes sense, I'm glad I asked as I would not have guessed that. Thanks for the the clarification.